File: smartfin-edu/src/types/market.ts
export interface MarketData {
  symbol: string;
  name: string;
  price: number;
  change: number;
  changePercent: number;
  volume: number;
  marketCap?: number;
  high52Week?: number;
  low52Week?: number;
  peRatio?: number;
  dividendYield?: number;
  lastUpdated: Date;
}

export interface HistoricalData {
  symbol: string;
  date: Date;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
  adjustedClose?: number;
}

export interface TechnicalIndicator {
  name: string;
  value: number;
  signal: 'BUY' | 'SELL' | 'HOLD';
  timestamp: Date;
}

export interface MarketIndex {
  symbol: string;
  name: string;
  value: number;
  change: number;
  changePercent: number;
  components?: string[];
}

export interface CryptocurrencyData extends MarketData {
  totalSupply?: number;
  circulatingSupply?: number;
  rank?: number;
}

export interface ForexData {
  baseCurrency: string;
  quoteCurrency: string;
  rate: number;
  change: number;
  changePercent: number;
  bid: number;
  ask: number;
  spread: number;
  lastUpdated: Date;
}

export interface WatchlistItem {
  id: string;
  userId: string;
  symbol: string;
  addedAt: Date;
  alertPrice?: number;
  notes?: string;
}

export interface Portfolio {
  id: string;
  userId: string;
  name: string;
  description?: string;
  totalValue: number;
  totalCost: number;
  totalGainLoss: number;
  totalGainLossPercent: number;
  positions: PortfolioPosition[];
  createdAt: Date;
  updatedAt: Date;
}

export interface PortfolioPosition {
  id: string;
  portfolioId: string;
  symbol: string;
  quantity: number;
  averageCost: number;
  currentPrice: number;
  marketValue: number;
  unrealizedGainLoss: number;
  unrealizedGainLossPercent: number;
  addedAt: Date;
  updatedAt: Date;
}

export interface MarketAlert {
  id: string;
  userId: string;
  symbol: string;
  condition: 'ABOVE' | 'BELOW' | 'CROSSES_ABOVE' | 'CROSSES_BELOW';
  targetPrice: number;
  currentPrice: number;
  isActive: boolean;
  triggered: boolean;
  triggeredAt?: Date;
  createdAt: Date;
}

export interface MarketNews {
  id: string;
  title: string;
  content: string;
  source: string;
  author?: string;
  publishedAt: Date;
  symbols?: string[];
  sentiment?: 'POSITIVE' | 'NEGATIVE' | 'NEUTRAL';
  importance: 'LOW' | 'MEDIUM' | 'HIGH';
  url?: string;
  imageUrl?: string;
}

File: backend/app/schemas/market.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class MarketDataResponse(BaseModel):
    symbol: str = Field(..., description="股票代码")
    name: str = Field(..., description="公司名称")
    price: float = Field(..., description="当前价格")
    open: float = Field(..., description="开盘价")
    high: float = Field(..., description="最高价")
    low: float = Field(..., description="最低价")
    prev_close: float = Field(..., description="前收盘价")
    change: float = Field(..., description="涨跌额")
    change_percent: float = Field(..., description="涨跌幅(%)")
    volume: int = Field(..., description="成交量")
    market_cap: Optional[int] = Field(None, description="市值")
    market: Optional[str] = Field(None, description="交易所")
    sector: Optional[str] = Field(None, description="行业")
    currency: str = Field(default="USD", description="货币")
    updated_at: str = Field(..., description="更新时间")

class HistoricalDataResponse(BaseModel):
    timestamp: str = Field(..., description="时间戳")
    open: float = Field(..., description="开盘价")
    high: float = Field(..., description="最高价")
    low: float = Field(..., description="最低价")
    close: float = Field(..., description="收盘价")
    volume: int = Field(..., description="成交量")

class SymbolSearchResponse(BaseModel):
    symbol: str = Field(..., description="股票代码")
    name: str = Field(..., description="公司名称")
    type: str = Field(..., description="资产类型")
    exchange: str = Field(..., description="交易所")
    market: str = Field(..., description="市场")

class BatchDataRequest(BaseModel):
    symbols: list[str] = Field(..., min_items=1, max_items=50, description="股票代码列表")